### ISSUE TITLE
When the depth of category sublevels > 3, ps_categoryproducts will not display only products in the current category

### ISSUE THREAD

**Issue by qwebdevel** [PrestaShop/PrestaShop#17902]
>Edit by Khouloud

**Steps to reproduce the issue:**
1. I created categories like in the screenshot
![image](https://user-images.githubusercontent.com/16067358/75761392-4c216f00-5d39-11ea-9487-7b5808cd9e26.png)
2. I created a product "g-shock" associated with the list of these categories
![image](https://user-images.githubusercontent.com/16067358/75761459-678c7a00-5d39-11ea-99c4-8231d8214bc3.png)
3. I created a product "edifice" associated with the list of these categories
![image](https://user-images.githubusercontent.com/16067358/75761595-a02c5380-5d39-11ea-9e78-05254e19633f.png)
4. In the FO, I have this
![image](https://user-images.githubusercontent.com/16067358/75761690-cd790180-5d39-11ea-8498-1a55bc5cb111.png)
5. In the FO => g-shock product details page
![image](https://user-images.githubusercontent.com/16067358/75761846-0d3fe900-5d3a-11ea-8366-fcf8b07e2c38.png)
The exact problem that the "Edifice" product is displayed in the `ps_categoryproducts` section, it shouldn't

-----------------------------------
I described a bug in products category i have a shop with many categories and sub categories when i add product and select home category / sub category / sub category module ps_categoryproducts show all products and not products in current sub category thanks.

**Comment by khouloudbelguith** [comment]
Hi @qwebdevel,

I did not manage to reproduce the issue with PS1.7.6.3 and ps_categoryproducts module v1.0.4.
I attached a screen record
https://drive.google.com/file/d/1kg9o4RX61cUgZjWlUKx3zyHGwwruzzwp/view
Thanks to check & feedback.

**Comment by qwebdevel** [comment]
Hi @khouloudbelguith just check url in product and url category is not the same and when you add product in category in ps_category module show all products not only the products in current category so i think this is not a module bug sorry for misunderstanding i don't explain the problem very well when you go in category url structure is /home/category/sub category/ when go to product url structure is /home/product_url need to be same like category

**Comment by khouloudbelguith** [comment]
@qwebdevel, it is an option in the ps_facetesearch module.
"Show products from subcategories" is enabled in your case, you need to disable it.
![image](https://user-images.githubusercontent.com/16067358/75697127-f0a99f80-5cac-11ea-9702-686e5b6a02ef.png)

Thanks to check & feedback.

**Comment by qwebdevel** [comment]
@khouloudbelguith just make new product add in /category/sub-category/sub-category and look url in domain is domainname.com/category/product_url after this with menu go to last sub category and go to product see url is domainname.com/category/sub-category/sub-category/product_url  so urls don't match i think this is not module problem.

**Comment by qwebdevel** [comment]
I will give you example i have watches site for my case is for casio so i make category and sub category /men/g-shock i add g-shock watch in /men/g-shock after this i make second category  and called /men/edifice i add edifice watch when i go to edifice watch or g-shock watch i see both in ps_category module which is in the end of product so i think there is problem related with product urls

**Comment by khouloudbelguith** [comment]
@qwebdevel,

1. I created categories like in the screenshot
![image](https://user-images.githubusercontent.com/16067358/75761392-4c216f00-5d39-11ea-9487-7b5808cd9e26.png)
2. I created a product "g-shock" associated with the list of these categories
![image](https://user-images.githubusercontent.com/16067358/75761459-678c7a00-5d39-11ea-99c4-8231d8214bc3.png)
3. I created a product "edifice" associated with the list of these categories
![image](https://user-images.githubusercontent.com/16067358/75761595-a02c5380-5d39-11ea-9e78-05254e19633f.png)
4. In the FO, I have this
![image](https://user-images.githubusercontent.com/16067358/75761690-cd790180-5d39-11ea-8498-1a55bc5cb111.png)
5. In the FO => g-shock product details page
![image](https://user-images.githubusercontent.com/16067358/75761846-0d3fe900-5d3a-11ea-8366-fcf8b07e2c38.png)
The exact problem that the "Edifice" product is displayed in the `ps_categoryproducts` section? is it?

Thanks to check & feedback.

**Comment by qwebdevel** [comment]
khouloudbelguith, i check this option just hide products in sub categories i don't need to hide products

**Comment by khouloudbelguith** [comment]
@qwebdevel,
>i check this option just hide products in sub categories i don't need to hide products

Yes!, This is another issue.

Can you confirm these steps: https://github.com/PrestaShop/PrestaShop/issues/17902#issuecomment-593853467 please?
Is this the real issue in your case?

Thanks!

**Comment by qwebdevel** [comment]
@khouloudbelguith, Yes is real issue :) in ps 1.6 show only product in current category where is product in ps 1.7 show all like mixed content when client looking for watch don't need to see panties or something else :)

**Comment by qwebdevel** [comment]
Yes this is the problem thank you for understanding and looking for problem deeply

**Comment by khouloudbelguith** [comment]
@qwebdevel,

Thanks for your report.
I manage to reproduce the issue with PS1.7.5.2 and PS1.7.6.3 using the `ps_categoryproducts` module v1.0.4.
With PS1.6.1.24 and [productscategory](https://github.com/PrestaShop/productscategory) module v1.8.1 it was ok.
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by Banhack** [comment]
We have the same problem, Does anyone have a solution?

**Comment by florine2623** [comment]
Tested with 8.1.x and the module ps_categoryproducts Products in the same category v1.0.7. I have the same behavior.
